src/dmd/clone.d

```diff
@@ -765,7 +765,10 @@ extern (C++) FuncDeclaration buildXtoHash(StructDeclaration sd, Scope* sc)
     const(char)* code =
         "size_t h = 0;" ~
         "foreach (i, T; typeof(p.tupleof))" ~
-        "    h = h * 33 + typeid(T).getHash(cast(const void*)&p.tupleof[i]);" ~
+        "    static if(is(T : const(Object))) " ~
```

Comment:
Would `is(T == class) || is(T == interface)` be better? It would avoid false positives for non-objects that implicitly convert to objects, such as:

https://run.dlang.io/is/xlZFQA
```d
struct S
{
    size_t toHash() const { return 3; }
    Object what;
    alias what this;
}

void main()
{
	S s;
    static assert(is(S : const(Object)));
    static assert(!is(S == class));
    assert(typeid(S).getHash(&s) == 3);
}
```

Reply:
OK, what about this:

```D
is(T* : const(Object)*)
```

I tested it against types with alias this, and it seems to work.

Reply:
`__traits(getLinkage)` works on classes and interfaces now https://github.com/dlang/dmd/commit/971777de5a19b960c343265efca3f9833ea071bc.

Reply:
Can you please add the test case from n8sh.

Reply:
@jacob-carlborg ok